diesel_cli/src/main.rs

```diff
@@ -186,6 +188,59 @@ fn convert_absolute_path_to_relative(target_path: &Path, mut current_path: &Path
     result.join(target_path.strip_prefix(current_path).unwrap())
 }
 
+fn run_infer_schema(matches: &ArgMatches) {
+    let database_url = database::database_url(matches);
+    let schema_name = None;
+    let schema = diesel_infer_schema::infer_schema_for_schema_name(&database_url, schema_name);
+    let mut out = String::new();
+    let mut indent = String::new();
+    let mut skip_space = false;
+    let mut last_char = ' ';
+    
+
+    for c in schema.as_str().chars() {
```

Comment:
Okay, IIUC this whole for-loop is to pretty print the tokens, right? Great job! In my tests you got about 80% of the formatting right in just 40 lines of code! Sadly it will look weird unless we get to 99% :)

Nevertheless, I'd either (a) put this into a separate function that takes `quote::Token` and a `W: io::Write` and make that _perfect_ for using with our `table!` macro, or (b) skip this completely and use a proven tool we don't need to maintain! :)

syn has a "pretty" feature, but that seems to use libsyntax' pretty printing… which is not that great. Can we maybe pipe our string output into rustfmt? That way it might even respect the users' project's rustfmt.toml.

Maybe @dtolnay, the great author of quote and syn, has some suggestions? :)